sktime/classification/deep_learning/transformer.py

```diff
@@ -0,0 +1,280 @@
+"""Pytorch Transformer Model."""
+
+import numpy as np
+
+from sktime.classification.deep_learning._pytorch import BaseDeepClassifierPytorch
+from sktime.utils.dependencies import _check_soft_dependencies
+
+if _check_soft_dependencies("torch", severity="none"):
+    import torch
+    from torch.utils.data import DataLoader, Dataset
+else:
+
+    class Dataset:
+        """Dummy class if torch is unavailable."""
+
+
+class TransformerClassifier(BaseDeepClassifierPytorch):
```

Comment:
there are multiple transformer based architectures for various tasks including classification - can we give it a more specific name, hence?

Reply:
Yes, you are right, it should not be given a generic name of `transformer`. As it is not the official transformer architecture, rather a proposed transformer architecture from an academic publication. The authors themselves call it "MVTS Transformer (Multivariate Time Series Transformer)".

Therefore I have renamed
- classes from `TransformerClassifier` to `MVTSTransformerClassifier`
- files from `transformer` to `mvts_transformer`

For reference: 
  - Paper: [A Transformer-based Framework for Multivariate Time Series Representation Learning](https://dl.acm.org/doi/abs/10.1145/3447548.3467401)
  - Code: [gzerveas/mvts_transformer](https://github.com/gzerveas/mvts_transformer)